### PR TITLE
Provisional Features Added + both bgimg plugins

### PR DESCRIPTION
**Pull request by namjohn920**
data-bgimg
data-bgimg-srcset

### CODE REVIEW

File: src/plugins/bgimg-srcset/bgimg-srcset.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+/*
+ * Variable and function definitions.
+ * These are global to the plugin - meaning that they will be initialized once per page,
+ * not once per instance of plugin on the page. So, this is a good place to define
+ * variables that are common to all instances of the plugin on a page.
+ */
+var $document = wb.doc,
+	$window = wb.win,
+	componentName = "wb-bgimg-srcset",
+	selector = "[data-bgimg-srcset]",
```
Comment (duboisp) [inline src/plugins/bgimg-srcset/bgimg-srcset.js]: Add provisional CSS class.

```
selector = ".provisional [data-bgimg-srcset]",
```

**Review by duboisp — CHANGES_REQUESTED** [review]

**Review by GormFrank — CHANGES_REQUESTED** [review]
Almost there!

File: site/pages/docs/ref/provisional-fr.hbs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+<div class="alert alert-warning">
+	<h2>Unstable feature</h2>
+	<p>To be used at <strong>your own risk</strong>. All functionalities described below can be removed in any subsequent minor/major release</p>
+	<p><small><a href="https://wet-boew.github.io/wet-boew-documentation/decision/7.html">Learn more about the design decision around provisional features.</a></small></p>
+	<p>The documentation and/or working examples for those features could be incomplete or not available.</p>
+</div>
+
+<h2>Feature availability</h2>
+
+<p>The following table describes until which version of GCWeb each provisional feature will be available. The features' availability will be re-evaluated at every release. A withdrawn provisional feature would be notified one version in advance.</p>
```
Comment (GormFrank) [inline site/pages/docs/ref/provisional-fr.hbs]:
```suggestion
<p>The following table describes until which version of WET-BOEW each provisional feature will be available. The features' availability will be re-evaluated at every release. A withdrawn provisional feature would be notified one version in advance.</p>
```

File: site/pages/docs/ref/provisional-fr.hbs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+<details>
+	<summary>Status description</summary>
+	<dl class="dl-horizontal">
+		<dt>&lt;not specified&gt;</dt>
+		<dd>provisional</dd>
+		<dt>Withdrawn</dt>
+		<dd>The feature are not available</dd>
+		<dt>Supported</dt>
+		<dd>The feature are included and supported by the GCWeb public API.</dd>
```
Comment (GormFrank) [inline site/pages/docs/ref/provisional-fr.hbs]:
```suggestion
		<dd>The feature are included and supported by the WET-BOEW public API.</dd>
```

File: site/pages/docs/ref/provisional-fr.hbs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+		<th>Description</th>
+		<th class="nowrap">Available in</th>
+		<th>Status</th>
+	</tr>
+</thead>
+<tbody>
+	<tr>
+		<th>CSS <code>.bg-cover</code></th>
+		<td>Background size cover</td>
+		<td>v5.1</td>
```
Comment (GormFrank) [inline site/pages/docs/ref/provisional-fr.hbs]:
```suggestion
		<td>v4.0.34</td>
```

File: site/pages/docs/ref/provisional-fr.hbs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+		<td></td>
+	</tr>
+	<tr>
+		<th>Plugin <code>[data-bgimg]</code></th>
+		<td>Take the URL value and set it as the background image. To be replaced by CSS4 selector.</td>
+		<td>v5.1</td>
```
Comment (GormFrank) [inline site/pages/docs/ref/provisional-fr.hbs]:
```suggestion
		<td>v4.0.34</td>
```

File: site/pages/docs/ref/provisional-fr.hbs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+		<td>Having <code>.provisional</code> class as a parent will be deprecated in July 2020 (approximately), which is 6 months after this feature was updated.</td>
+	</tr>
+	<tr>
+		<th>Plugin <code>[data-bgimg-srcset]</code></th>
+		<td>Detects the change in screen width and replace the background image accordingly</td>
+		<td>v5.1</td>
```
Comment (GormFrank) [inline site/pages/docs/ref/provisional-fr.hbs]:
```suggestion
		<td>v4.0.34</td>
```

**Review by GormFrank — CHANGES_REQUESTED** [review]

**Review by GormFrank — APPROVED** [review]

**Comment by GormFrank** [comment]
We'll need to add a blurb in the provisional features in GCWeb to indicate that those features have been moved. That and also we'll need to change the "experimental" wording to "provisional" again in GCWeb.

File: src/plugins/bgimg-srcset/bgimg-srcset.js [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+	selectImage = function() {
+		var screenWidth = window.innerWidth,
+			optimizedLink = {},
+			i_len = ids.length;
+
+		for ( var i = 0; i < i_len; i++ ) {
+			var optimizedSize = Infinity;
+			for ( var j = 0; j < inputs[ ids[ i ] ].length; j++ ) {
+				if ( inputs[ ids[ i ] ][ j ][ 1 ] >= screenWidth ) {
+					if ( optimizedSize > inputs[ ids[ i ] ][ j ][ 1 ] ) {
```
Comment (GormFrank) [inline src/plugins/bgimg-srcset/bgimg-srcset.js]:
```suggestion
					var inputImg = inputs[ ids[ i ] ][ j ];
					if ( optimizedSize > inputImg[ 1 ]  ) {
```

**Review by GormFrank — CHANGES_REQUESTED** [review]
Add variables and replace the code by those vars

**Review by GormFrank — APPROVED** [review]

File: theme/_provisional.scss [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+	.bg-gctheme {
+		background-color:#355688;
+	}
+
+	.bg-gctheme.well.header-rwd {
+		background-color: #26374a;
+	}
+
+	.bg-img-hdng {
```
Comment (GormFrank) [inline theme/_provisional.scss]: This one should not have been brought over here. It is for a different provisional feature.

**Review by GormFrank — CHANGES_REQUESTED** [review]

File: site/pages/docs/ref/provisional-en.hbs [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+<pre><code>&lt;div class="provisional bg-cover well" data-bgimg="../../demos/tabs/img/investinourfuture.jpg"&gt;
+	&lt;div class="well mrgn-tp-md mrgn-bttm-md"&gt;
+		&lt;h4 class="mrgn-tp-md"&gt;Heading&lt;/h4&gt;
+		&lt;p&gt;Secondary title&lt;/p&gt;
+	&lt;/div&gt;
+&lt;/div&gt;</code></pre>
+
+<h3 id="well-header-rwd-bgimg">Responsive background image using [data-bgimg-srcset] plugin</h3>
+<p>Background pictures will change as screen width size changes.</p>
```
Comment (duboisp) [inline site/pages/docs/ref/provisional-en.hbs]: Can you add a link on how to a publisher could configure the data-bgimg-srcset?
And add 2-4 code sample of possible configuration.

And please replicate that instruction in French too

File: src/plugins/bgimg/bgimg.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+/*
+ * Variable and function definitions.
+ * These are global to the plugin - meaning that they will be initialized once per page,
+ * not once per instance of plugin on the page. So, this is a good place to define
+ * variables that are common to all instances of the plugin on a page.
+ */
+var $document = wb.doc,
+	componentName = "wb-bgimg",
+	selector = ".provisional[data-bgimg], .provisional [data-bgimg]",
```
Comment (duboisp) [inline src/plugins/bgimg/bgimg.js]: Need to be also compatible with the deprecated markup that we have introduced in GCWeb. 
We would deprecate the old markup and we would remove it in 6 month. That would require a note in teh GCWeb provisional page.

File: theme/_provisional.scss [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+	.bg-darker.well.header-rwd h1 {
+		border-bottom-color:#6f6f6f;
+	}
+
+	.bg-gctheme {
+		background-color:#355688;
+	}
+
+	.bg-gctheme.well.header-rwd {
```
Comment (duboisp) [inline theme/_provisional.scss]: This is to be removed here, but we need to keep it in GCWeb.

File: theme/_provisional.scss [hunk 12]
```diff
@@ -0,0 +1,30 @@
+.provisional{
+	.bg-darker {
+		background-color:#000;
+	}
+
+	.bg-darker.well.header-rwd h1 {
+		border-bottom-color:#6f6f6f;
+	}
+
+	.bg-gctheme {
```
Comment (duboisp) [inline theme/_provisional.scss]: This is to be removed here, but we need to keep it in GCWeb.

**Comment by GormFrank** [comment]
Pre-approved during technical review, as long as change requests are resolved.

File: theme/_provisional.scss [hunk 13]
```diff
@@ -0,0 +1,30 @@
+.provisional{
+	.bg-darker {
+		background-color:#000;
+	}
+
+	.bg-darker.well.header-rwd h1 {
```
Comment (duboisp) [inline theme/_provisional.scss]: This is to be removed here, but we need to keep it in GCWeb.

File: theme/_provisional.scss [hunk 14]
```diff
@@ -0,0 +1,30 @@
+.provisional{
+	.bg-darker {
```
Comment (duboisp) [inline theme/_provisional.scss]: This is to be removed here, but we need to keep it in GCWeb.

**Review by duboisp — CHANGES_REQUESTED** [review]

**Review by GormFrank — APPROVED** [review]

**Review by duboisp — APPROVED** [review]

**Pull request merged**